File: internal/content/testdata/bench/javascript/nc-subtree-of-another.js
function is_subtree(root, sub) {
    function same(a, b) {
        if (a === null && b === null) return true;
        if (a === null || b === null || a.val !== b.val) return false;
        return same(a.left, b.left) && same(a.right, b.right);
    }
    if (sub === null) return true;
    function dfs(node) {
        if (node === null) return false;
        if (same(node, sub)) return true;
        return dfs(node.left) || dfs(node.right);
    }
    return dfs(root);
}

File: internal/content/testdata/bench/rust/nc-subtree-of-another.rs
use std::rc::Rc;
use std::cell::RefCell;

fn is_subtree(root: Option<Rc<RefCell<TreeNode>>>, sub: Option<Rc<RefCell<TreeNode>>>) -> bool {
    fn same(a: &Option<Rc<RefCell<TreeNode>>>, b: &Option<Rc<RefCell<TreeNode>>>) -> bool {
        match (a, b) {
            (None, None) => true,
            (Some(x), Some(y)) => {
                let xb = x.borrow();
                let yb = y.borrow();
                if xb.val != yb.val {
                    return false;
                }
                same(&xb.left, &yb.left) && same(&xb.right, &yb.right)
            }
            _ => false,
        }
    }
    fn dfs(node: &Option<Rc<RefCell<TreeNode>>>, sub: &Option<Rc<RefCell<TreeNode>>>) -> bool {
        match node {
            None => false,
            Some(n) => {
                if same(node, sub) {
                    return true;
                }
                let nb = n.borrow();
                dfs(&nb.left, sub) || dfs(&nb.right, sub)
            }
        }
    }
    if sub.is_none() {
        return true;
    }
    dfs(&root, &sub)
}
